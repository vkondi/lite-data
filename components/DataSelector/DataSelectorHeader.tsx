import React from 'react';
import { FormControl, InputLabel } from '@mui/material';
import styles from './DataSelectorHeader.module.css';

const DataSelectorHeader: React.FC = () => {
  return (
    <div className={styles.container}>
      <FormControl variant="outlined" sx={{ flex: 1, borderRight: '1px solid #e0e0e0' }}>
        <InputLabel>Data Type</InputLabel>
      </FormControl>
      <FormControl variant="outlined" sx={{ flex: 1,borderRight: '1px solid #e0e0e0' }}>
        <InputLabel>Field Name</InputLabel>
      </FormControl>
      <div className={styles.deleteIconSpacer} data-testid="delete-icon-spacer"></div>
    </div>
  );
};

export default DataSelectorHeader;