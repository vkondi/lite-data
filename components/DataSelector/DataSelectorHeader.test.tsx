import { render, screen } from "@testing-library/react";
import { describe, it, expect } from "vitest";
import DataSelectorHeader from "./DataSelectorHeader";

describe("DataSelectorHeader", () => {
  it("renders without crashing", () => {
    const { container } = render(<DataSelectorHeader />);
    expect(container).toMatchSnapshot();
  });

  it("renders the correct labels", () => {
    const { getByText } = render(<DataSelectorHeader />);
    expect(getByText("Data Type")).toBeInTheDocument();
    expect(getByText("Field Name")).toBeInTheDocument();
  });

  it("renders the delete icon spacer", () => {
    render(<DataSelectorHeader />);
    const fields = screen.queryAllByTestId("delete-icon-spacer");
    expect(fields.length).toBe(1);
  });
});
