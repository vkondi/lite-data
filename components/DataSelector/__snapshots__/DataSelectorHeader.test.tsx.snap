// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DataSelectorHeader > renders without crashing 1`] = `
<div>
  <div
    class="_container_66254c"
  >
    <div
      class="MuiFormControl-root css-vav2vq-MuiFormControl-root"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-19qnlrw-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="false"
      >
        Data Type
      </label>
    </div>
    <div
      class="MuiFormControl-root css-vav2vq-MuiFormControl-root"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-19qnlrw-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="false"
      >
        Field Name
      </label>
    </div>
    <div
      class="_deleteIconSpacer_66254c"
      data-testid="delete-icon-spacer"
    />
  </div>
</div>
`;
